app/models/event.rb

```diff
@@ -29,7 +29,17 @@ def notify_users
     EventNotifications.perform_async(id)
   end
 
-  def self.notify_users_available_todays
-    Event.find_by(start_date: DateTime.now).each(&:users_available)
+  def attend(user_id)
+    Attendance.create(event_id: id, user_id: user_id) unless existing?(user_id)
+  end
+
+  def cancel(user_id)
+    Attendance.find_by(user_id: user_id, event_id: id).destroy
+  end
+
+  private
+
+  def existing?(user_id)
+    Attendance.find_by(user_id: user_id, event_id: id).present?
```

Comment:
same thing in these two about `find_by`.

You can do `where(...).destroy_all` or `where(...).first.destroy`